azure-mgmt-documentdb/src/main/java/com/microsoft/azure/management/documentdb/implementation/DatabaseAccountImpl.java

```diff
@@ -277,4 +255,99 @@ private void addLocationsForCreateUpdateParameters(
+                        self.failoverPolicies.clear();
+                        self.hasFailoverPolicyChanges = false;
+                        return manager().databaseAccounts().getByResourceGroupAsync(
+                                resourceGroupName(),
+                                name()
+                        ).repeatWhen(new Func1<Observable<? extends java.lang.Void>, Observable<?>>() {
+                            @Override
+                            public Observable<?> call(Observable<? extends Void> observable) {
+                                return observable.delay(5, TimeUnit.SECONDS);
+                            }
+                        })
+                        .filter(new Func1<DatabaseAccount, Boolean>() {
+                            @Override
+                            public Boolean call(DatabaseAccount databaseAccount) {
+                                if (databaseAccount.id() == null
+                                        || databaseAccount.id().length() == 0) {
+                                    return false;
+                                }
+
+                                if (isAFinalProvisioningState(databaseAccount.inner().provisioningState())) {
+                                    for (Location location : databaseAccount.readableReplications()) {
+                                        if (!isAFinalProvisioningState(location.provisioningState())) {
+                                            return false;
+                                        }
+
+                                    }
+                                } else {
+                                    return false;
+                                }
+
+                                return true;
+                            }
+                        })
+                        .first();
+
+                    }
+                });
+    }
+
+    private void ensureFailoverIsInitialized() {
+        if (this.isInCreateMode()) {
+            return;
+        }
+
+        if (!this.hasFailoverPolicyChanges) {
+            this.failoverPolicies.clear();
+            FailoverPolicyInner[] policyInners = new FailoverPolicyInner[this.inner().failoverPolicies().size()];
+            this.inner().failoverPolicies().toArray(policyInners);
+            Arrays.sort(policyInners, new Comparator<FailoverPolicyInner>() {
+                @Override
+                public int compare(FailoverPolicyInner o1, FailoverPolicyInner o2) {
+                    return o1.failoverPriority().compareTo(o2.failoverPriority());
+                }
+            });
+
+            for (int i = 0; i < policyInners.length; i++) {
+                this.failoverPolicies.add(policyInners[i]);
+            }
+
+            this.hasFailoverPolicyChanges = true;
+        }
+    }
+
+    private boolean isAFinalProvisioningState(String state) {
+        return state.equalsIgnoreCase("succeeded");

```

Comment:
It's possible that the provisioning state ends up "Failed" or "Canceled"? Probably don't want to wait forever in those cases.